### ISSUE TITLE
`RegExp.prototype.unicode` and the `/u` flag

### ISSUE THREAD

**Issue by mathiasbynens** [google/traceur-compiler#370]
How would you like to see this implemented in Traceur?

I guess we’d need a regex parser in JavaScript, and then transform any regular expressions using the `/u` flag into backwards-compatible regexes without the `/u` flag.

[_Regenerate_](http://mths.be/regenerate) is a tool that can be used to generate regular expressions based on a set of code points, symbols (i.e. strings containing only a single Unicode character) or ranges of code points/symbols. After the parsing stage, it could be used to rewrite the regular expressions if they contain any astral symbols.

Also `.` would need to be rewritten when the `/u` flag is enabled. From http://mathiasbynens.be/notes/javascript-unicode#regex:

> When the `u` flag is set, `.` is equivalent to the following backwards-compatible regular expression pattern:
> 
> ``` js
> >> regenerate()
>      .addRange(0x0, 0x10FFFF) // all Unicode code points
>      .remove(  // minus `LineTerminator`s (http://ecma-international.org/ecma-262/5.1/#sec-7.3):
>        0x000A, // Line Feed <LF>
>        0x000D, // Carriage Return <CR>
>        0x2028, // Line Separator <LS>
>        0x2029  // Paragraph Separator <PS>
>      )
>      .toString();
> '[\0-\x09\x0B\x0C\x0E-\u2027\u202A-\uD7FF\uDC00-\uFFFF]|[\uD800-\uDBFF][\uDC00-\uDFFF]|[\uD800-\uDBFF]'
> 
> >> /foo(?:[\0-\x09\x0B\x0C\x0E-\u2027\u202A-\uD7FF\uDC00-\uFFFF]|[\uD800-\uDBFF][\uDC00-\uDFFF]|[\uD800-\uDBFF])bar/u.test('foo\u{1D306}bar')
> true
> ```

In other words, `.` could be replaced by `(?:[\0-\x09\x0B\x0C\x0E-\u2027\u202A-\uD7FF\uDC00-\uFFFF]|[\uD800-\uDBFF][\uDC00-\uDFFF]|[\uD800-\uDBFF])`.

---

Situations that require looking at code points rather than UCS-2/UTF-16 code units in order to support the full Unicode character set:
- the regex is case-insensitive
- the regex contains a character class
- the regex uses `.`
- the regex uses a quantifier
